proposals/infra/INF-0002-root-signature-in-clang.md

```diff
@@ -0,0 +1,717 @@
+  GlobalRootSignature = 0x2,
+};
+
+enum class RootSignatureFlags : uint32_t {
+  None = 0,
+  AllowInputAssemblerInputLayout = 0x1,
+  DenyVertexShaderRootAccess = 0x2,
+  DenyHullShaderRootAccess = 0x4,
+  DenyDomainShaderRootAccess = 0x8,
+  DenyGeometryShaderRootAccess = 0x10,
+  DenyPixelShaderRootAccess = 0x20,
+  AllowStreamOutput = 0x40,
+  LocalRootSignature = 0x80,
+  DenyAmplificationShaderRootAccess = 0x100,
+  DenyMeshShaderRootAccess = 0x200,
+  CBVSRVUAVHeapDirectlyIndexed = 0x400,
+  SamplerHeapDirectlyIndexed = 0x800,
+  AllowLowTierReservedHwCbLimit = 0x80000000,
+  ValidFlags = 0x80000fff
+};
+enum class RootParameterType : uint32_t {
+  DescriptorTable = 0,
+  Constants32Bit = 1,
+  CBV = 2,
+  SRV = 3,
+  UAV = 4,
+  MaxValue = 4
+};
+
+
+
+enum class ComparisonFunc : uint32_t {
+  Never = 1,
+  Less = 2,
+  Equal = 3,
+  LessEqual = 4,
+  Greater = 5,
+  NotEqual = 6,
+  GreaterEqual = 7,
+  Always = 8
+};
+enum class Filter : uint32_t {
+  MIN_MAG_MIP_POINT = 0,
+  MIN_MAG_POINT_MIP_LINEAR = 0x1,
+  MIN_POINT_MAG_LINEAR_MIP_POINT = 0x4,
+  MIN_POINT_MAG_MIP_LINEAR = 0x5,
+  MIN_LINEAR_MAG_MIP_POINT = 0x10,
+  MIN_LINEAR_MAG_POINT_MIP_LINEAR = 0x11,
+  MIN_MAG_LINEAR_MIP_POINT = 0x14,
+  MIN_MAG_MIP_LINEAR = 0x15,
+  ANISOTROPIC = 0x55,
+  COMPARISON_MIN_MAG_MIP_POINT = 0x80,
+  COMPARISON_MIN_MAG_POINT_MIP_LINEAR = 0x81,
+  COMPARISON_MIN_POINT_MAG_LINEAR_MIP_POINT = 0x84,
+  COMPARISON_MIN_POINT_MAG_MIP_LINEAR = 0x85,
+  COMPARISON_MIN_LINEAR_MAG_MIP_POINT = 0x90,
+  COMPARISON_MIN_LINEAR_MAG_POINT_MIP_LINEAR = 0x91,
+  COMPARISON_MIN_MAG_LINEAR_MIP_POINT = 0x94,
+  COMPARISON_MIN_MAG_MIP_LINEAR = 0x95,
+  COMPARISON_ANISOTROPIC = 0xd5,
+  MINIMUM_MIN_MAG_MIP_POINT = 0x100,
+  MINIMUM_MIN_MAG_POINT_MIP_LINEAR = 0x101,
+  MINIMUM_MIN_POINT_MAG_LINEAR_MIP_POINT = 0x104,
+  MINIMUM_MIN_POINT_MAG_MIP_LINEAR = 0x105,
+  MINIMUM_MIN_LINEAR_MAG_MIP_POINT = 0x110,
+  MINIMUM_MIN_LINEAR_MAG_POINT_MIP_LINEAR = 0x111,
+  MINIMUM_MIN_MAG_LINEAR_MIP_POINT = 0x114,
+  MINIMUM_MIN_MAG_MIP_LINEAR = 0x115,
+  MINIMUM_ANISOTROPIC = 0x155,
+  MAXIMUM_MIN_MAG_MIP_POINT = 0x180,
+  MAXIMUM_MIN_MAG_POINT_MIP_LINEAR = 0x181,
+  MAXIMUM_MIN_POINT_MAG_LINEAR_MIP_POINT = 0x184,
+  MAXIMUM_MIN_POINT_MAG_MIP_LINEAR = 0x185,
+  MAXIMUM_MIN_LINEAR_MAG_MIP_POINT = 0x190,
+  MAXIMUM_MIN_LINEAR_MAG_POINT_MIP_LINEAR = 0x191,
+  MAXIMUM_MIN_MAG_LINEAR_MIP_POINT = 0x194,
+  MAXIMUM_MIN_MAG_MIP_LINEAR = 0x195,
+  MAXIMUM_ANISOTROPIC = 0x1d5
+};
+enum class ShaderVisibility : uint32_t {
+  All = 0,
+  Vertex = 1,
+  Hull = 2,
+  Domain = 3,
+  Geometry = 4,
+  Pixel = 5,
+  Amplification = 6,
+  Mesh = 7,
+  MaxValue = 7
+};
+enum class StaticBorderColor : uint32_t {
+  TransparentBlack = 0,
+  OpaqueBlack = 1,
+  OpaqueWhite = 2,
+  OpaqueBlackUint = 3,
+  OpaqueWhiteUint = 4
+};
+enum class TextureAddressMode : uint32_t {
+  Wrap = 1,
+  Mirror = 2,
+  Clamp = 3,
+  Border = 4,
+  MirrorOnce = 5
+};
+
+struct ContainerRootDescriptor {
```

Comment:
Many of the structs defined below have initialized default values, but there are some with members that are missing values.  Was this intentional?

Example:
ContainerRootDescriptor initializes RegisterSpace but not the others.
ContainerDescriptorRange is missing inits

Reply:
Not all members have default value.
Added the missing RegisterSpace init.